openshift_performance/ci/scripts/large_object_list/large_object_list.sh

```diff
@@ -0,0 +1,176 @@
+function create_namespaces_and_secrets() {
+
+  for ((i=0; i<$expected_namespaces; i++)); do
+      my_namespace_command='oc new-project --skip-config-write "${my_namespace}${i}" > /dev/null && echo -e "\n\nCreated project ${my_namespace}${i}" &>> object-list-test.log &'
+      eval $my_namespace_command
+      for ((j=0; j<$secrets_per_namespace; j++)); do
+         echo "
+         apiVersion: v1
+         kind: Secret
+         metadata:
+           labels:
+             test: listobject
+           namespace: ${my_namespace}${i}
+           name: ${my_namespace}${i}${my_secret_name}${j}
+         " | oc apply -f - &>> object-list-test.log &
+      done
+      sleep 5
+  done
+}
+
+# function to delete namespaces (and secrets)
+function delete_namespaces() {
+  my_expected_namespaces=$1
+  COUNTER=0  
+
+  for ((i=0; i<$my_expected_namespaces; i++)); do
+      my_command='oc delete project "${my_namespace}${i}" >> object-list-test.log &'
+      eval $my_command
+  done
+
+  existing_obj=$(oc get $my_object_type -A| grep $my_namespace | wc -l)
+  while [ $existing_obj -ne 0 ]; do
+    sleep 5
+    existing_obj=$(oc get $my_object_type -A | grep $my_namespace | wc -l | xargs )
+    echo "Waiting for $my_object_type to be deleted: $existing_obj still exist"
+    COUNTER=$((COUNTER + 1))
+    if [ $COUNTER -ge 60 ]; then
+      echo "$existing_obj $my_object_type are still not deleted after 5 minutes"
+      exit 1
+    fi
+  done
+  echo "All $my_object_type are deleted"
+}
+
+
+# Start of the test
+# Step 1. Check the cluster for objects created by previous execution of this test. If those objects exist, remove them.
+echo "Checking the cluster for test objects..."
+check_cluster=$(oc get $my_object_type -A| grep $my_namespace | wc -l)
+if [ $check_cluster -ne 0 ]; then
+   echo "Removing test objects from the cluster before test execution"
+   delete_namespaces $check_cluster
+fi
+
+# Step 2. Start the log file
+echo -e "\nWriting subprocess logs to ./object-list-test.log"
+if [ -e object-list-test.log ]; then
+	echo "object-list-test.log exists, subprocess logs will be appended"
+fi
+
+echo -e "\n===========================================" >> object-list-test.log &
+echo "$(date) - NEW RUN" >> object-list-test.log &
+echo ""
+
+
+# Step 3. Create the namespaces and the secrets. The output is printed to a log file. Also, we are calculating test object creation time.
+echo "$(date) - Namespace and secret creation starting..."
+cycle_start_time=`date +%s`
+create_namespaces_and_secrets
+sleep 5
+
+echo "$(date) - Namespace and secret creation complete."
+echo -e "\nNamespace and secret creation complete." >> object-list-test.log &
+cycle_end_time=`date +%s` 
+total_cycle_time=$((cycle_end_time - cycle_start_time))
+echo "Total time for create cycle: $total_cycle_time s."
+
+
+# Pause to allow the system to catch up
+echo -e "\nSleeping for 5 minutes..."
+echo -e "\nSleeping for 5 minutes..." >> object-list-test.log &
```

Comment:
use '| tee object-list-test.log' can output to console at the same time write to a log file